src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMiningTests.cs

```diff
@@ -31,16 +32,20 @@ public void MiningAndPropagatingPOS_MineBlockStakeAtInsufficientHeightError()
         {
             using (NodeBuilder nodeBuilder = NodeBuilder.Create(this))
             {
-                CoreNode node = nodeBuilder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet();
+                var network = new StratisRegTest();
+
+                CoreNode node = nodeBuilder.CreateStratisPosNode(network).NotInIBD().WithWallet();
                 nodeBuilder.StartAll();
 
                 // Mine two blocks (OK).
                 TestHelper.MineBlocks(node, 2);
 
                 // Mine another block after LastPOWBlock height (Error).
-                nodeBuilder.Nodes[0].FullNode.Network.Consensus.LastPOWBlock = 2;
+                var lastPowBlock = node.FullNode.Network.Consensus.LastPOWBlock;
```

Comment:
@bokobza this was the main issue here, we need to set the value back after the Assert has completed.